packages/headless/src/controllers/tab/headless-tab.test.ts

```diff
@@ -38,10 +39,42 @@ describe('Tab', () => {
       };
       initTab();
 
+      const action = setDefaultConstantQueryFilter(expression);
+      expect(engine.actions).toContainEqual(action);
+    });
+
+    it('calls #registerConstantQuery if isActive is true and actaulValueWasChanged is false', () => {
+      props = {
+        options: {
+          expression,
+        },
+        initialState: {
+          isActive: true,
+        },
+      };
+      engine.state.defaultConstantQueryFilter.actualValueWasChanged = false;
+      initTab();
+
       const action = updateAdvancedSearchQueries({cq: expression});
       expect(engine.actions).toContainEqual(action);
     });
 
+    it("doesn't call #registerConstantQuery if isActive is true and actaulValueWasChanged is true", () => {
```

Comment:
There should be no `actaulValueWasChanged` state in the controller. This should all be handled by the reducer.

Controllers are just meant to simplify interacting with the engine. One should be able to get the same behaviour by dispatching actions.

When we starting adding stateful logic to controllers, it now means you either must use the controller, or you have to reimplement the same checks when using the actions. When the logic is in the reducer, you can do either and get the same experience.